core/ast_node.js

```diff
@@ -474,6 +471,31 @@ Blockly.ASTNode.prototype.getOutAstNodeForBlock_ = function(block) {
   }
 };
 
+/**
+ * Find the first editable field or input with a connection on a given block.
+ * @param {Blockly.BlockSvg} block The source block of the current location.
+ * @return {Blockly.ASTNode} An ast node pointing to the first field or input.
+ * Null if there are no editable fields or inputs with connections on the block.
+ * @private
+ */
+Blockly.ASTNode.prototype.findFirstFieldOrInput_ = function(block) {
+  var inputs = block.inputList;
+  for (var i = 0; i < inputs.length; i++) {
+    var input = inputs[i];
```

Comment:
Generally we use this form:
`  for (var i = 0, input; input = inputs[i]; i++) {`
Here and in the j loop below.

Reply:
Fixed